entity-framework/core/what-is-new/ef-core-10.0/breaking-changes.md

```diff
@@ -178,6 +182,137 @@ await context.Blogs.ExecuteUpdateAsync(s =>
 });
 ```
 
+<a name="compiled-model-private-methods"></a>
+
+### Compiled models now throw exception for value converters with private methods
+
+[Tracking Issue #35033](https://github.com/dotnet/efcore/issues/35033)
+
+#### Old behavior
+
+Previously, when using value converters with compiled models (using `dotnet ef dbcontext optimize`), EF would reference the converter type and everything worked correctly.
+
+```c#
+public sealed class BooleanToCharConverter : ValueConverter<bool, char>
+{
+    public static readonly BooleanToCharConverter Default = new();
+
+    public BooleanToCharConverter()
+        : base(v => ConvertToChar(v), v => ConvertToBoolean(v))
+    {
+    }
+
+    private static char ConvertToChar(bool value) // Private method
+    {
+        return value ? 'Y' : 'N';
+    }
+
+    private static bool ConvertToBoolean(char value) // Private method
+    {
+        return value == 'Y';
+    }
+}
+```
+
+#### New behavior
+
+Starting with EF Core 10.0, EF generates code that directly references the conversion methods themselves. If these methods are private, compilation will fail.
+
+#### Why
+
+This change was necessary to support NativeAOT.
+
+#### Mitigations
+
+Make the methods referenced by value converters public or internal instead of private.
+
+<a name="complex-type-column-uniquification"></a>
+
+### Complex type column names are now uniquified
+
+[Tracking Issue #4970](https://github.com/dotnet/EntityFramework.Docs/issues/4970)
+
+#### Old behavior
+
+Previously, when mapping complex types to table columns, if multiple properties in different complex types had the same column name, they would silently share the same column.
```

Comment:
How does this relate to the breaking change just below - are they dups? In other words, if the full path is now used (the change below), how is it possible to for multiple properties in different complex types to have the same column name (i.e. when would appending the number be necessary)?